### PR TITLE
docs(aiplatform): Add GenAI samples for C#

### PR DESCRIPTION
**Pull request by meteatamel**
- Edits to `PredictTextPromptSample`

- Add: `PredictTextEmbeddingsSample` `PredictChatPromptSample`, `PredictTextExtractionSample`, `PredictCodeChatSample`, `PredictCodeGenerationUnitTestSample`, `PredictCodeCompletionCommentSample`, `PredictTextSummarizationSample`, `PredictCodeCompletionTestFunctionSample`, `PredictTextClassificationSample`, `PredictCodeGenerationFunctionSample`, `PredictTextSentimentSample`

### CODE REVIEW

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            messages = new[]
+            {
+                new
+                {
+                    author = "user",
+                    content = prompt
+                }
+            }
+        });
+        //var instance = wkt::Value.Parser.ParseJson(instanceJson);
```
Comment (holtskinner) [inline aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:69]:
```suggestion
        // var instance = wkt::Value.Parser.ParseJson(instanceJson);
```

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            },
+            messages = new[]
+            {
+                new
+                {
+                    author = "user",
+                    content = prompt
+                }
+            }
+        });
```
Comment (holtskinner) [inline aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:68]: I like the idea of showing both methods, but it seems a little overwhelming to have so much in this one sample. Maybe stick to one or the other?

**Review by holtskinner — APPROVED** [review]

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs (hunk 2 above)
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:68]: Hi @holtskinner, I agree it's a little overwhelming but Protobuf is really impossible to construct on your own and JSON-to-Protobuf is brittle. I wanted to have one sample where people see both and decide which one to use on their own. In other samples, I'll probably stick with Protobuf only but it'd be useful to have one sample to show both ways, so people can make the link between the two. What do you think?

Comment (Rishabh-V) [reply aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:68]: Good to get the context on this. Sharing my thoughts - Given it is a sample, which should  be clear, easy to understand and simple I would prefer to use the Protobuf directly in sample.  If we want to show the other two approaches we can possibly add it in the documentation page or add the corresponding approaches in readme.md and paste its link in the sample. That way we can keep this sample simple and clear and also provide alternatives. Ones the library is out with correct protos, we will need to revisit this, so we can also add a TODO.   This applies to both instances and parameters.

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        // Initialize client that will be used to send requests.
+        // This client only needs to be created
+        // once, and can be reused for multiple requests.
+        var client = new PredictionServiceClientBuilder
+        {
+            Endpoint = $"{locationId}-aiplatform.googleapis.com"
+        }.Build();
+
+        // Configure the parent resource
+        var endpoint = EndpointName.FromProjectLocationPublisherModel(projectId, locationId, publisher, model).ToString();
```
Comment (Rishabh-V) [inline aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:43]: We can just create an `endpoint`  without `.ToString()` and use the overload of `Predict` that uses `EndpointName`

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                    { "topK", new wkt::Value { NumberValue = 40 } }
+                }
+            }
+        };
+
+        // Make the request
+        var response = client.Predict(endpoint, instances, parameters);
+
+        // Parse and return the response
+        var content = response.Predictions[0].StructValue.Fields["candidates"].ListValue.Values[0].StructValue.Fields["content"].StringValue;
```
Comment (Rishabh-V) [inline aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:145]: Ah! This is hard.

**Review by Rishabh-V — COMMENTED** [review]
Hey Mete, I am reviewing your PR as part of training to become a full reviewer. There will be an additional review by Amanda after this. I have shared my thoughts just so that I can see and learn what Amanda thinks about those suggestions and I will learn with that. Please do not make any code changes until Amanda requests any changes. Thank you.

File: aiplatform/api/AIPlatform.Samples.Tests/PredictChatPromptTests.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Xunit;
+
+[Collection(nameof(AIPlatformFixture))]
+public class PredictChatPromptTest
```
[the remark this answers is not in the document]
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples.Tests/PredictChatPromptTests.cs:20]: Yes, I just followed the other test file. Let me know what's the preferred way.

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START aiplatform_sdk_chat]
+
+using Google.Cloud.AIPlatform.V1;
```
[the remark this answers is not in the document]
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:19]: I'm on Mac with VSCode, do you know an extension that can help? I tried 'C# Sort Usings' but doesn't seem to work

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    {
+        // Initialize client that will be used to send requests.
+        // This client only needs to be created
+        // once, and can be reused for multiple requests.
+        var client = new PredictionServiceClientBuilder
+        {
+            Endpoint = $"{locationId}-aiplatform.googleapis.com"
+        }.Build();
+
+        // Configure the parent resource
```
[the remark this answers is not in the document]
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:42]: You mean the missing full stops in the comments?

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs (hunk 4 above)
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:145]: I tried to match the Java version of the samples that return the response..I'm parsing the content, to show people how to do it, as it's not obvious at all.

File: aiplatform/api/AIPlatform.Samples.Tests/PredictChatPromptTests.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Rishabh-V) [reply aiplatform/api/AIPlatform.Samples.Tests/PredictChatPromptTests.cs:20]: Let's see what Amanda has to say.

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (Rishabh-V) [reply aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:19]: Haven't used VS Code for a while. However, the top few results look promising - https://marketplace.visualstudio.com/search?term=Sort%20usings&target=VSCode&category=All%20categories&sortBy=Relevance

(In Visual Studio, organizing usings is offered out of the box)

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (Rishabh-V) [reply aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:42]: Yep. We try to follow that in our review. Please wait for Amanda.

**Review by Rishabh-V — COMMENTED** [review]
Hey Mete, Please wait for Amanda to do a review.  Things may change. Thanks for your patience.

File: aiplatform/api/AIPlatform.Samples.Tests/PredictChatPromptTests.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples.Tests/PredictChatPromptTests.cs]: Looking at the other test files in other projects, I think it should be PredictChatPromptTest.cs. That's what I'm doing now.

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs]: Yes I think we should have full stops, added them

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs (hunk 3 above)
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs]: Good catch. I removed the toString

File: aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs (hunk 4 above)
Comment (meteatamel) [reply aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs:145]: Actually, Java version doesn't return anything..so I agree with you. We should just parse what we need and return that instead. That's what I'm doing now.

**Comment by meteatamel** [comment]
I'm going to close this and open new PRs for each sample

**Pull request closed without merging**